src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Arm/AdvSimd.PlatformNotSupported.cs

```diff
@@ -0,0 +1,377 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.Intrinsics.Arm
+{
+    /// <summary>
+    /// This class provides access to the ARM AdvSIMD hardware instructions via intrinsics
+    /// </summary>
+    [CLSCompliant(false)]
+    public abstract class AdvSimd : ArmBase
```

Comment:
This probably was asked and discussed before - but why `AdvSimd` and other ISA classes are marked as abstract and not static instead? How inheritance of AdvSimd from ArmBase is going to be used?

Reply:
`static` classes can't use inheritance.

Inheriting means that dotting off the `AdvSimd` also gives you access to ISAs that are logically supported as a subset (so `AdvSimd.LeadingZeroCount` can be done, for example).

Reply:
Thank you, this makes sense.

Reply:
@tannergooding unrelated question: Why X86.Lzcnt is not a subclass of X86.Bmi1 ?

Reply:
Because it is its own CPUID bit and it is not classified as part of `Bmi1` by all x86 hardware.

More specifically, AMD has the `ABM` feature set (which is just `POPCNT` and `LZCNT`) which is the same CPUID bit as used for `LZCNT` on Intel. `POPCNT`, on Intel, is technically considered part of `SSE4.2`, but it also has its own CPUID bit (which AMD sets on machines that have the `ABM` flag set).

So ultimately, these two instructions (`LZCNT` and `POPCNT`) are their own 'ISAs" even if the individual architectures will bundle them differently.